js/ui/number_box/number_box.mask.js

```diff
@@ -243,6 +243,10 @@ const NumberBoxMask = NumberBoxBase.inherit({
             if(this._parsedValue < 0 || 1 / this._parsedValue === -Infinity) {
                 this._revertSign(e);
                 this._setTextByParsedValue();
+
+                if(this.option('valueChangeEvent').split(' ').includes('input')) {
```

Comment:
1. It's not necessary to split - string has [includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) method
2. Let's move this to a variable `shouldTriggerInputEvent` to simplify future debug